src/coreclr/tools/SuperFileCheck/Program.cs

```diff
@@ -0,0 +1,760 @@
+
+        /// <summary>
+        /// Get the method name from the method declaration.
+        /// </summary>
+        static string GetMethodName(MethodDeclarationSyntax methodDecl)
+        {
+            var methodName = methodDecl.Identifier.ValueText;
+
+            var typeArity = methodDecl.TypeParameterList?.ChildNodes().Count();
+            if (typeArity > 0)
+            {
+                throw new SuperFileCheckException($"'{methodName}' has generic parameters which are not supported.");
+            }
+
+            return $"{methodName}(*)";
+        }
+
+        /// <summary>
+        /// Try to get the enclosing type declaration syntax of the given node.
+        /// </summary>
+        static TypeDeclarationSyntax? TryGetEnclosingTypeDeclaration(SyntaxNode node)
+        {
+            return node.Ancestors().OfType<TypeDeclarationSyntax>().FirstOrDefault();
+        }
+
+        /// <summary>
+        /// Get all the acestoral enclosing type declaration syntaxes of the given node.
+        /// </summary>
+        static TypeDeclarationSyntax[] GetEnclosingTypeDeclarations(SyntaxNode node)
+        {
+            return node.Ancestors().OfType<TypeDeclarationSyntax>().ToArray();
+        }
+
+        /// <summary>
+        /// Try to get an enclosing type name from the given syntax node.
+        /// </summary>
+        static string GetTypeName(TypeDeclarationSyntax typeDecl)
+        {
+            var typeName = typeDecl.Identifier.ValueText;
+
+            var typeArity = typeDecl.TypeParameterList?.ChildNodes().Count();
+            if (typeArity > 0)
+            {
+                throw new SuperFileCheckException($"'{typeName}' has generic parameters which are not supported.");
+            }
+
+            return typeName;
+        }
+
+        /// <summary>
+        /// Get the method's fully qualified enclosing namespace and type name.
+        /// </summary>
+        static string GetFullyQualifiedEnclosingTypeName(MethodDeclarationSyntax methodDecl)
+        {
+            var qualifiedTypeName = String.Empty;
+
+            var typeDecl = TryGetEnclosingTypeDeclaration(methodDecl);
+            if (typeDecl == null)
+            {
+                return qualifiedTypeName;
+            }
+
+            qualifiedTypeName = GetTypeName(typeDecl);
+
+            var typeDecls = GetEnclosingTypeDeclarations(typeDecl);
+            for (var i = 0; i < typeDecls.Length; i++)
+            {
+                typeDecl = typeDecls[i];
+                qualifiedTypeName = $"{GetTypeName(typeDecl)}+{qualifiedTypeName}";
+            }
+
+            var namespaceDecl = typeDecl.Ancestors().OfType<NamespaceDeclarationSyntax>().FirstOrDefault();
+            if (namespaceDecl != null)
+            {
+                var identifiers =
+                    namespaceDecl.DescendantTokens().Where(x => x.IsKind(SyntaxKind.IdentifierToken)).Select(x => x.ValueText);
+                var namespac = String.Join(".", identifiers);
```

Comment:
I don't see where the <code>namespac</code> variable is used - am I missing something?

Reply:
You are not missing anything - looks like I have the same `String.Join(".", identifiers);` right below it, I guess I was trying to factor it out?. I'll remove the `namespac`.